packages/contracts/test/diamond/facets/DirectGovernanceFarmerFacet.t.sol

```diff
@@ -569,20 +604,38 @@ contract DirectGovernanceFarmerTest is LocalTestHelper {
 
         vm.mockCall(
             depositZapAddress,
-            abi.encodeWithSelector(
-                IDepositZap.remove_liquidity_one_coin.selector
-            ),
-            abi.encode(100e18)
+            abi.encodeWithSelector(IDepositZap.remove_liquidity.selector),
+            abi.encode([100e18, 99e18, 98e18, 97e18])
         );
-
-        uint256 tokenAmount = directGovernanceFarmer.withdraw(
+        vm.expectEmit(
+            true,
+            true,
+            true,
+            true,
+            address(directGovernanceFarmerFacet)
+        );
+        emit WithdrawAll(
+            userAddress,
             1,
-            address(token0)
+            [uint256(100e18), uint256(99e18), uint256(98e18), uint256(97e18)]
+        );
+        uint256[4] memory tokenAmounts = directGovernanceFarmerFacet.withdrawId(
+            1
         );
-        assertEq(tokenAmount, 100e18);
+        vm.stopPrank();
+        assertEq(tokenAmounts[0], 100e18);
+        assertEq(tokenAmounts[1], 99e18);
+        assertEq(tokenAmounts[2], 98e18);
+        assertEq(tokenAmounts[3], 97e18);
+        assertEq(dollar.balanceOf(userAddress), 100e18);
+        assertEq(token0.balanceOf(userAddress), 100e18);
+        assertEq(token1.balanceOf(userAddress), 100e18);
+        assertEq(token2.balanceOf(userAddress), 100e18);
     }
 
-    function testIsIdIncludedReturnTrueIfIdIsInTheList() public {
+    // END Multiple
+
+    /*function testIsIdIncludedReturnTrueIfIdIsInTheList() public {
```

Comment:
because this can be also set an assertion in depositSingle test when staking happens the user gets into the array ID list, maybe we want to have it alone or inside the [depositSingle ](https://github.com/ubiquity/ubiquity-dollar/pull/841/files#diff-6140f6576af81654e444ba3b5bddd1fc61cfaf0a5670769f670bd2fb4ecb3527R185)test maybe not?

Reply:
It's not clear how [isIdIncluded()](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol#L161) is related to staking in terms of testing.

[isIdIncluded()](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol#L161) simply checks whether a uint value is in the list or not. This is basically what [these tests](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/test/diamond/facets/DirectGovernanceFarmerFacet.t.sol#L638-L663) are checking. Why can't we use the same assertions but using the facet's [method](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol#L161) now?

Reply:
You get an ID when you stake, the deposits functions all returns a [stakingShareId](https://github.com/ubiquity/ubiquity-dollar/pull/841/files#diff-4fe442d71f05b6caaf7a99183584f8e4f2b48eba36403391c2e8517a276a7551R95), the lists that you mention are the listed depositors so isIdIncluded checks if depositor is on the list, so it's a helper function, we could have an assertion in the deposit list test, (which in fact tests that the helper is working) or we could have the same test as the original contract (doing the same but like re-doing)

Reply:
Each test must check it's own method thus we separate concerns. If we assert `isIdIncluded()` in the `depositSingle()` method we violate separation of concerns. 

Basically we could assert all of the methods in a single test but we don't do it because this is hard to maintain hence we divide all tests to assert small pieces of logic and features.

Reply:
revamping this test to meet the it's criteria

Reply:
in terms of representation the depositors are forwarded as uint values thus getting arrayed on the list, this helper function a nice utility to facilitate the inclusion and returning a boolean to the value include